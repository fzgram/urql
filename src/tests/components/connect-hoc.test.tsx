import React from 'react';
import ConnectHOC from '../../components/connect-hoc';
import renderer from 'react-test-renderer';

describe('Connect HOC', () => {
  it('should wrap its component argument with connect', () => {
    const Comp = args => <div {...args} />;
    const Wrapped = ConnectHOC()(Comp);
    // @ts-ignore
    const component = renderer.create(
      // @ts-ignore
      <Wrapped />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
