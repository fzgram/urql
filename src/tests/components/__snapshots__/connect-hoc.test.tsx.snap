// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connect HOC should wrap its component argument with connect 1`] = `
<div
  data={null}
  error={null}
  fetching={false}
  loaded={false}
  refetch={[Function]}
/>
`;
